### DISCUSSION TITLE
Why do you need us, the end user, to create a developer app in order to use the plugin?

### DISCUSSION THREAD

**Discussion by DanForever** [fredemmott/StreamDeck-Discord#34] [Q&A] [answered]
Are you not able to create a single app yourself, without the need to pretend to be an app developer? It seems like there's something missing in the design of the UX somewhere.

To be clear, this seems to be a convention with a lot of stream deck apps (the Spotify plugin also had me jump through similar hoops), I just don't understand why it's necessary.

**Comment by fredemmott** [comment, marked as answer]
1. Discord will not approve any new apps using the local API
2. Unapproved apps can use the API, but can only be used by up to 50 people, which need individually, manually, adding through Discord's web UI
3. Even if this wasn't the case, Discord's local app authentication[is non-standard and not fit for purpose](https://support.discord.com/hc/en-us/community/posts/360038666632-IPC-authentication), and effectively would need me to give anyone the ability to write apps impersonating me, and if they wrote malware, my personal Discord account could be banned

**Reply by fredemmott** [reply]
I also absolutely agree it would be much better if there were a single, pre-configured app, but it's not possible unless Discord change their APIs and restrictions.

**Reply by DanForever** [reply]
Wait, so Discord has a publicly accessible "local api" (presumably "local" means modify user settings, rather than modify server settings?) but they won't allow anyone to publish an app that uses it? O_o

**Reply by fredemmott** [reply]
'local' means "runs on your computer"

They used to allow apps to be published, and they allow apps that were previously approved to keep working. For example, the legacy logitech software (pre g-hub) has a discord plugin for lighting up when there's a message. Additionally, Discord's 'now playing' , and 'join me in this game' features are built on top of this local API - approval is possible if you just use those parts, but not the rest.

I could technically make this work by impersonating the logitech app, but that's a gray area I don't want to go near.

**Reply by DanForever** [reply]
or when you say local, you mean using "localhost" as the redirect? Presumably it'd be possible to get around that with some webhosting somewhere that can act as an intermediary between the user and discord

**Reply by fredemmott** [reply]
Specifically, some operations like self-deafen and self-mute are only available over named pipes (windows) and unix sockets (macos), not over a network - those *must* be done by software running on your computer, not on some webhost.

**Reply by DanForever** [reply]
Gosh, how lame is discord. Thank you for you prompt answers, this has been enlightening, if disheartening

**Reply by fredemmott** [reply]
For completeness: there is also a *new* local API, that it is possible to get approved for - it looks like it offers the same features, but it actually doesn't, and isn't usable here.

Self-mute and self-deafen via the new local api only affect calls made by the app - this is intended for games that use discord for their in-game chat, and offer in-game mute/deafen. These only mute/deafen the in-game voice chat, not anything else you're doing in discord.

**Reply by fredemmott** [reply]
> Spotify

Barraider's Discord is probably the best place to get a definitive answer to this. As I understand it, it's a different problem: Spotify limit an app to a fixed number of requests per minute/hour, and won't increase it. There's too many users, so he can't fit them in an app any more.

For example with made up numbers, if the limit is 100/hour and you have 200 peak users, it's not going to work well.
